kolibri2zim/scraper.py

```diff
@@ -880,18 +932,21 @@ def sanitize_inputs(self):
         channel_meta = self.db.get_channel_metadata(self.channel_id)
 
         # input  & metadata sanitation
-        period = datetime.datetime.now().strftime("%Y-%m")
+        period = datetime.datetime.now(datetime.UTC).strftime("%Y-%m")
         if self.fname:
             # make sure we were given a filename and not a path
-            self.fname = Path(self.fname.format(period=period))
-            if Path(self.fname.name) != self.fname:
-                raise ValueError(f"filename is not a filename: {self.fname}")
+            fname_path = Path(self.fname.format(period=period)).resolve()
```

Comment:
I don't understand this change. `this.fname` should be a filename, not a path to a filename. Resolving it makes no sense because it will be used as filename inside the `output_dir`. Testing for `is_file()` doesn't make sense either because this name should not be a path on the FS (which returns `False`).

Reply:
I agree that resolving it is useless.
The issue is that the type checker complains that self.fname might be a `Path` (because it does not realized that it is transformed as a `Path` only afterward, probably because it does not detect that `sanitize_inputs` is called only once).
I got confused by `if Path(self.fname.name) != self.fname:` meaning.
I fixed it to a more equivalent version.